packages/addon/lua/gelly/mods/blood-mod/init.lua

```diff
@@ -96,21 +158,33 @@ end
 hook.Add(
 	"GellyXDamage",
 	"gelly.builtin.blood-mod",
-	function(_, attacker, position, force, damage, type)
+	function(victim, attacker, position, force, damage, type)
 		if
-			not attacker:IsPlayer()
-			and not attacker:IsNPC()
-			and attacker:GetClass() ~= "prop_ragdoll"
+			not victim:IsValid() or
+			(not victim:IsPlayer() and
+			not victim:IsNPC() and
+			not victim:IsRagdoll())
 		then
 			return
 		end
-		-- 8194 is the combination used in M9K for bullets
-		if type == 8194 then
-			type = 2
+
+		local bloodColor = BLOOD_COLOR_RED
+
+		if victim:GetBloodColor() ~= nil and
+			victim:GetBloodColor() ~= DONT_BLEED and
```

Comment:
This effectively means we ignore when an entity was configured to explicitly not bleed
Is that behaviour we want?

Possibly using the blood colour returned by the entity directly is ideal (and then exiting early if we dont have a material defined for it). Maybe still with the fallback from nil to red?